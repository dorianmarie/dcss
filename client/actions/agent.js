import {
  GET_AGENT_ERROR,
  GET_AGENT_SUCCESS,
  GET_AGENTS_ERROR,
  GET_AGENTS_SUCCESS,
  SET_AGENT_ERROR,
  SET_AGENT_SUCCESS
} from './types';

export let getAgent = id => async dispatch => {
  try {
    const res = await (await fetch(`/api/agents/${id}`)).json();

    if (res.error) {
      throw res;
    }
    const { agent } = res;

    dispatch({ type: GET_AGENT_SUCCESS, agent });
    return agent;
  } catch (error) {
    dispatch({ type: GET_AGENT_ERROR, error });
    return null;
  }
};

export let setAgent = (id, params) => async dispatch => {
  try {
    if (Object.values(params).length) {
      const body = JSON.stringify(params);
      const res = await (await fetch(`/api/agents/${id}`, {
        method: 'PUT',
        headers: {
          'Content-Type': 'application/json'
        },
        body
      })).json();

      if (res.error) {
        throw res;
      }
      const { agent } = res;
      dispatch({ type: SET_AGENT_SUCCESS, agent });
      return agent;
    }
    return null;
  } catch (error) {
    dispatch({ type: SET_AGENT_ERROR, error });
    return null;
  }
};

export let createAgent = params => async dispatch => {
  try {
    if (Object.values(params).length) {
      const body = JSON.stringify(params);
      const res = await (await fetch(`/api/agents`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json'
        },
        body
      })).json();

      if (res.error) {
        throw res;
      }
      const { agent } = res;
      dispatch({ type: GET_AGENT_SUCCESS, agent });
      return agent;
    }
    return null;
  } catch (error) {
    dispatch({ type: GET_AGENT_ERROR, error });
    return null;
  }
};

export let getAgents = filter => async dispatch => {
  try {
    const url =
      filter === 'is_active' ? '/api/agents/is_active' : '/api/agents';

    const res = await (await fetch(url)).json();

    if (res.error) {
      throw res;
    }

    const { agents = [] } = res;
    dispatch({ type: GET_AGENTS_SUCCESS, agents });
    return agents;
  } catch (error) {
    dispatch({ type: GET_AGENTS_ERROR, error });
    return null;
  }
};
