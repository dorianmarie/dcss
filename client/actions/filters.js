import { SET_FILTER_SCENARIOS_IN_USE } from './types';

export let setFilterScenariosInUse = scenariosInUse => dispatch => {
  dispatch({ type: SET_FILTER_SCENARIOS_IN_USE, scenariosInUse });
};
