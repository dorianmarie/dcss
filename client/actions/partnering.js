import { GET_PARTNERING_ERROR, GET_PARTNERING_SUCCESS } from './types';

export let getPartnering = () => async (dispatch, getState) => {
  const state = getState();

  if (state.partnering.length) {
    return state.partnering;
  }

  try {
    const res = await (await fetch('/api/partnering')).json();

    if (res.error) {
      throw res;
    }

    const { partnering } = res;

    dispatch({ type: GET_PARTNERING_SUCCESS, partnering });
    return partnering;
  } catch (error) {
    dispatch({ type: GET_PARTNERING_ERROR, error });
    return null;
  }
};
