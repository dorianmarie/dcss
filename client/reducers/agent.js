import {
  GET_AGENT_SUCCESS,
  SET_AGENT_SUCCESS,
  GET_AGENTS_SUCCESS
} from '@actions/types';
import { agentInitialState } from './initial-states';

export const agent = (state = agentInitialState, action) => {
  const { agent, type } = action;
  switch (type) {
    case GET_AGENT_SUCCESS:
    case SET_AGENT_SUCCESS: {
      return {
        ...state,
        ...agent
      };
    }
    default:
      return state;
  }
};

export const agents = (state = [], action) => {
  const { agent, agents, type } = action;
  switch (type) {
    case GET_AGENT_SUCCESS:
    case SET_AGENT_SUCCESS: {
      const index = state.findIndex(({ id }) => id === agent.id);
      if (index !== -1) {
        state[index] = {
          ...state[index],
          ...agent
        };
      } else {
        state.push(agent);
      }
      return [...state].sort((a, b) => a.id < b.id);
    }
    case GET_AGENTS_SUCCESS: {
      const seen = {
        /*
        agent.id: index
         */
      };
      return [...state, ...agents]
        .reduce((accum, agent) => {
          // If we've already seen this agent, then it was already
          // in "state" and we should replace it with the newer version.
          const index = seen[agent.id];
          if (index !== undefined) {
            accum[index] = agent;
          } else {
            seen[agent.id] = accum.length;
            accum.push(agent);
          }
          return accum;
        }, [])
        .sort((a, b) => a.id < b.id);
    }
    default:
      return state;
  }
};

export const agentsById = (state = {}, action) => {
  const { agent, agents, type } = action;
  switch (type) {
    case GET_AGENT_SUCCESS:
    case SET_AGENT_SUCCESS: {
      return {
        ...state,
        [agent.id]: {
          ...agent
        }
      };
    }
    case GET_AGENTS_SUCCESS: {
      const agentsById = agents.reduce((accum, agent) => {
        accum[agent.id] = agent;
        return accum;
      }, {});
      return {
        ...state,
        ...agentsById
      };
    }
    default:
      return state;
  }
};
