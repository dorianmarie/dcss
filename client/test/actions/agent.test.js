import { v4 as uuid } from 'uuid';
import assert from 'assert';
import {
  createMockStore,
  createMockConnectedStore,
  fetchImplementation,
  makeById,
  state
} from '../bootstrap';

import * as actions from '../../actions/agent';
import * as types from '../../actions/types';
import Storage from '../../util/Storage';
jest.mock('../../util/Storage');

const error = new Error('something unexpected happened on the server');
const original = JSON.parse(JSON.stringify(state));
let mockStore;
let store;

beforeAll(() => {
  (window || global).fetch = jest.fn();
  Storage.has = jest.fn();
  Storage.delete = jest.fn();
});

afterAll(() => {
  jest.restoreAllMocks();
  Storage.has.mockRestore();
  Storage.delete.mockRestore();
});

beforeEach(() => {
  mockStore = createMockStore({});
  store = createMockConnectedStore({});
  fetch.mockImplementation(() => {});
  Storage.has.mockImplementation(() => true);
  Storage.delete.mockImplementation(() => {});
});

afterEach(() => {
  jest.resetAllMocks();
  Storage.has.mockReset();
  Storage.delete.mockReset();
});

describe('GET_AGENT_SUCCESS', () => {
  describe('getAgent', () => {
    let agent = { ...state.agents[0], name: 'Fake Agent' };

    test('Receives a agent', async () => {
      fetchImplementation(fetch, 200, { agent });
      const returnValue = await store.dispatch(actions.getAgent(1));
      expect(fetch.mock.calls.length).toBe(1);
      expect(fetch.mock.calls[0]).toMatchInlineSnapshot(`
        Array [
          "/api/agents/1",
        ]
      `);
      expect(returnValue).toEqual(agent);

      await mockStore.dispatch(actions.getAgent(1));
      expect(mockStore.getActions()).toMatchSnapshot();
    });
  });

  describe('createAgent', () => {
    let agent = { ...state.agents[0], name: 'Fake Agent' };

    test('Receives a agent', async () => {
      fetchImplementation(fetch, 200, { agent });
      const returnValue = await store.dispatch(actions.createAgent(agent));
      expect(fetch.mock.calls.length).toBe(1);
      expect(fetch.mock.calls[0]).toMatchInlineSnapshot(`
        Array [
          "/api/agents",
          Object {
            "body": "{\\"id\\":1,\\"created_at\\":\\"2021-02-25T17:31:33.826Z\\",\\"updated_at\\":\\"2021-02-25T20:09:04.999Z\\",\\"deleted_at\\":null,\\"is_active\\":true,\\"title\\":\\"Emoji Analysis\\",\\"name\\":\\"Fake Agent\\",\\"description\\":\\"Detects the presense of an emoji character in your text\\",\\"endpoint\\":\\"ws://emoji-analysis-production.herokuapp.com\\",\\"configuration\\":{\\"bar\\":\\"2\\",\\"baz\\":\\"c\\",\\"foo\\":\\"false\\"},\\"interaction\\":{\\"id\\":1,\\"name\\":\\"ChatPrompt\\",\\"description\\":\\"It will appear as an option for scenario authors to include in chat discussions within multi-participant scenarios. It receives participant chat messages.\\",\\"created_at\\":\\"2021-02-25T15:09:05.001302-05:00\\",\\"deleted_at\\":null,\\"updated_at\\":null},\\"owner\\":{\\"id\\":999,\\"email\\":\\"super@email.com\\",\\"roles\\":[\\"participant\\",\\"super_admin\\",\\"facilitator\\",\\"researcher\\"],\\"is_super\\":true,\\"username\\":\\"superuser\\",\\"is_anonymous\\":false,\\"personalname\\":\\"Super User\\"},\\"self\\":{\\"id\\":148,\\"email\\":null,\\"roles\\":null,\\"is_super\\":false,\\"username\\":\\"ebe565050b31cbb4e7eacc39b23e2167\\",\\"lastseen_at\\":\\"2021-02-25T13:08:57.323-05:00\\",\\"is_anonymous\\":true,\\"personalname\\":\\"Emoji Analysis\\",\\"single_use_password\\":false}}",
            "headers": Object {
              "Content-Type": "application/json",
            },
            "method": "POST",
          },
        ]
      `);
      expect(returnValue).toEqual(agent);

      await mockStore.dispatch(actions.createAgent(agent));
      expect(mockStore.getActions()).toMatchSnapshot();
    });

    test('Empty params', async () => {
      fetchImplementation(fetch, 200, { agent });
      const returnValue = await store.dispatch(actions.createAgent({}));
      expect(fetch.mock.calls.length).toBe(0);
      expect(returnValue).toEqual(null);

      await mockStore.dispatch(actions.setAgent(1, {}));
      expect(mockStore.getActions()).toMatchSnapshot();
    });
  });
});

describe('GET_AGENT_ERROR', () => {
  let agent = { ...state.agents[0], name: 'Fake Agent' };

  describe('getAgent', () => {
    test('Receives an error', async () => {
      fetchImplementation(fetch, 200, { error });
      const returnValue = await store.dispatch(actions.getAgent(1));
      expect(fetch.mock.calls.length).toBe(1);
      expect(fetch.mock.calls[0]).toMatchInlineSnapshot(`
        Array [
          "/api/agents/1",
        ]
      `);
      expect(returnValue).toEqual(null);

      await mockStore.dispatch(actions.getAgent(1));
      expect(mockStore.getActions()).toMatchSnapshot();
    });
  });

  describe('createAgent', () => {
    test('Receives an error', async () => {
      fetchImplementation(fetch, 200, { error });
      const returnValue = await store.dispatch(actions.createAgent(agent));
      expect(fetch.mock.calls.length).toBe(1);
      expect(fetch.mock.calls[0]).toMatchInlineSnapshot(`
        Array [
          "/api/agents",
          Object {
            "body": "{\\"id\\":1,\\"created_at\\":\\"2021-02-25T17:31:33.826Z\\",\\"updated_at\\":\\"2021-02-25T20:09:04.999Z\\",\\"deleted_at\\":null,\\"is_active\\":true,\\"title\\":\\"Emoji Analysis\\",\\"name\\":\\"Fake Agent\\",\\"description\\":\\"Detects the presense of an emoji character in your text\\",\\"endpoint\\":\\"ws://emoji-analysis-production.herokuapp.com\\",\\"configuration\\":{\\"bar\\":\\"2\\",\\"baz\\":\\"c\\",\\"foo\\":\\"false\\"},\\"interaction\\":{\\"id\\":1,\\"name\\":\\"ChatPrompt\\",\\"description\\":\\"It will appear as an option for scenario authors to include in chat discussions within multi-participant scenarios. It receives participant chat messages.\\",\\"created_at\\":\\"2021-02-25T15:09:05.001302-05:00\\",\\"deleted_at\\":null,\\"updated_at\\":null},\\"owner\\":{\\"id\\":999,\\"email\\":\\"super@email.com\\",\\"roles\\":[\\"participant\\",\\"super_admin\\",\\"facilitator\\",\\"researcher\\"],\\"is_super\\":true,\\"username\\":\\"superuser\\",\\"is_anonymous\\":false,\\"personalname\\":\\"Super User\\"},\\"self\\":{\\"id\\":148,\\"email\\":null,\\"roles\\":null,\\"is_super\\":false,\\"username\\":\\"ebe565050b31cbb4e7eacc39b23e2167\\",\\"lastseen_at\\":\\"2021-02-25T13:08:57.323-05:00\\",\\"is_anonymous\\":true,\\"personalname\\":\\"Emoji Analysis\\",\\"single_use_password\\":false}}",
            "headers": Object {
              "Content-Type": "application/json",
            },
            "method": "POST",
          },
        ]
      `);
      expect(returnValue).toEqual(null);

      await mockStore.dispatch(actions.createAgent(agent));
      expect(mockStore.getActions()).toMatchSnapshot();
    });
  });
});

describe('SET_AGENT_SUCCESS', () => {
  let agent = { ...state.agents[0], name: 'Fake Agent' };

  describe('setAgent', () => {
    test('Empty params', async () => {
      fetchImplementation(fetch, 200, { agent });
      const returnValue = await store.dispatch(actions.setAgent(1, {}));
      expect(fetch.mock.calls.length).toBe(0);
      expect(returnValue).toEqual(null);

      await mockStore.dispatch(actions.setAgent(1, {}));
      expect(mockStore.getActions()).toMatchSnapshot();
    });

    test('Receives a agent', async () => {
      fetchImplementation(fetch, 200, { agent });
      const returnValue = await store.dispatch(actions.setAgent(1, agent));
      expect(fetch.mock.calls.length).toBe(1);
      expect(fetch.mock.calls[0]).toMatchInlineSnapshot(`
        Array [
          "/api/agents/1",
          Object {
            "body": "{\\"id\\":1,\\"created_at\\":\\"2021-02-25T17:31:33.826Z\\",\\"updated_at\\":\\"2021-02-25T20:09:04.999Z\\",\\"deleted_at\\":null,\\"is_active\\":true,\\"title\\":\\"Emoji Analysis\\",\\"name\\":\\"Fake Agent\\",\\"description\\":\\"Detects the presense of an emoji character in your text\\",\\"endpoint\\":\\"ws://emoji-analysis-production.herokuapp.com\\",\\"configuration\\":{\\"bar\\":\\"2\\",\\"baz\\":\\"c\\",\\"foo\\":\\"false\\"},\\"interaction\\":{\\"id\\":1,\\"name\\":\\"ChatPrompt\\",\\"description\\":\\"It will appear as an option for scenario authors to include in chat discussions within multi-participant scenarios. It receives participant chat messages.\\",\\"created_at\\":\\"2021-02-25T15:09:05.001302-05:00\\",\\"deleted_at\\":null,\\"updated_at\\":null},\\"owner\\":{\\"id\\":999,\\"email\\":\\"super@email.com\\",\\"roles\\":[\\"participant\\",\\"super_admin\\",\\"facilitator\\",\\"researcher\\"],\\"is_super\\":true,\\"username\\":\\"superuser\\",\\"is_anonymous\\":false,\\"personalname\\":\\"Super User\\"},\\"self\\":{\\"id\\":148,\\"email\\":null,\\"roles\\":null,\\"is_super\\":false,\\"username\\":\\"ebe565050b31cbb4e7eacc39b23e2167\\",\\"lastseen_at\\":\\"2021-02-25T13:08:57.323-05:00\\",\\"is_anonymous\\":true,\\"personalname\\":\\"Emoji Analysis\\",\\"single_use_password\\":false}}",
            "headers": Object {
              "Content-Type": "application/json",
            },
            "method": "PUT",
          },
        ]
      `);
      expect(returnValue).toEqual(agent);

      await mockStore.dispatch(actions.setAgent(1, agent));
      expect(mockStore.getActions()).toMatchSnapshot();
    });

    test('Receives a agent, updates scenario', async () => {
      store = createMockConnectedStore({
        ...original,
        scenario: {
          ...original.scenario,
          agents: [...original.agents]
        }
      });

      fetchImplementation(fetch, 200, { agent });
      const returnValue = await store.dispatch(actions.setAgent(1, agent));
      expect(fetch.mock.calls.length).toBe(1);
      expect(fetch.mock.calls[0]).toMatchInlineSnapshot(`
        Array [
          "/api/agents/1",
          Object {
            "body": "{\\"id\\":1,\\"created_at\\":\\"2021-02-25T17:31:33.826Z\\",\\"updated_at\\":\\"2021-02-25T20:09:04.999Z\\",\\"deleted_at\\":null,\\"is_active\\":true,\\"title\\":\\"Emoji Analysis\\",\\"name\\":\\"Fake Agent\\",\\"description\\":\\"Detects the presense of an emoji character in your text\\",\\"endpoint\\":\\"ws://emoji-analysis-production.herokuapp.com\\",\\"configuration\\":{\\"bar\\":\\"2\\",\\"baz\\":\\"c\\",\\"foo\\":\\"false\\"},\\"interaction\\":{\\"id\\":1,\\"name\\":\\"ChatPrompt\\",\\"description\\":\\"It will appear as an option for scenario authors to include in chat discussions within multi-participant scenarios. It receives participant chat messages.\\",\\"created_at\\":\\"2021-02-25T15:09:05.001302-05:00\\",\\"deleted_at\\":null,\\"updated_at\\":null},\\"owner\\":{\\"id\\":999,\\"email\\":\\"super@email.com\\",\\"roles\\":[\\"participant\\",\\"super_admin\\",\\"facilitator\\",\\"researcher\\"],\\"is_super\\":true,\\"username\\":\\"superuser\\",\\"is_anonymous\\":false,\\"personalname\\":\\"Super User\\"},\\"self\\":{\\"id\\":148,\\"email\\":null,\\"roles\\":null,\\"is_super\\":false,\\"username\\":\\"ebe565050b31cbb4e7eacc39b23e2167\\",\\"lastseen_at\\":\\"2021-02-25T13:08:57.323-05:00\\",\\"is_anonymous\\":true,\\"personalname\\":\\"Emoji Analysis\\",\\"single_use_password\\":false}}",
            "headers": Object {
              "Content-Type": "application/json",
            },
            "method": "PUT",
          },
        ]
      `);
      expect(returnValue).toEqual(agent);

      await mockStore.dispatch(actions.setAgent(1, agent));
      expect(mockStore.getActions()).toMatchSnapshot();
    });
  });
});

describe('SET_AGENT_ERROR', () => {
  let agent = { ...state.agents[0], name: 'Fake Agent' };

  describe('setAgent', () => {
    test('Receives an error', async () => {
      fetchImplementation(fetch, 200, { error });
      const returnValue = await store.dispatch(actions.setAgent(1, agent));
      expect(fetch.mock.calls.length).toBe(1);
      expect(fetch.mock.calls[0]).toMatchInlineSnapshot(`
        Array [
          "/api/agents/1",
          Object {
            "body": "{\\"id\\":1,\\"created_at\\":\\"2021-02-25T17:31:33.826Z\\",\\"updated_at\\":\\"2021-02-25T20:09:04.999Z\\",\\"deleted_at\\":null,\\"is_active\\":true,\\"title\\":\\"Emoji Analysis\\",\\"name\\":\\"Fake Agent\\",\\"description\\":\\"Detects the presense of an emoji character in your text\\",\\"endpoint\\":\\"ws://emoji-analysis-production.herokuapp.com\\",\\"configuration\\":{\\"bar\\":\\"2\\",\\"baz\\":\\"c\\",\\"foo\\":\\"false\\"},\\"interaction\\":{\\"id\\":1,\\"name\\":\\"ChatPrompt\\",\\"description\\":\\"It will appear as an option for scenario authors to include in chat discussions within multi-participant scenarios. It receives participant chat messages.\\",\\"created_at\\":\\"2021-02-25T15:09:05.001302-05:00\\",\\"deleted_at\\":null,\\"updated_at\\":null},\\"owner\\":{\\"id\\":999,\\"email\\":\\"super@email.com\\",\\"roles\\":[\\"participant\\",\\"super_admin\\",\\"facilitator\\",\\"researcher\\"],\\"is_super\\":true,\\"username\\":\\"superuser\\",\\"is_anonymous\\":false,\\"personalname\\":\\"Super User\\"},\\"self\\":{\\"id\\":148,\\"email\\":null,\\"roles\\":null,\\"is_super\\":false,\\"username\\":\\"ebe565050b31cbb4e7eacc39b23e2167\\",\\"lastseen_at\\":\\"2021-02-25T13:08:57.323-05:00\\",\\"is_anonymous\\":true,\\"personalname\\":\\"Emoji Analysis\\",\\"single_use_password\\":false}}",
            "headers": Object {
              "Content-Type": "application/json",
            },
            "method": "PUT",
          },
        ]
      `);
      expect(returnValue).toEqual(null);

      await mockStore.dispatch(actions.setAgent(1, agent));
      expect(mockStore.getActions()).toMatchSnapshot();
    });
  });
});

describe('GET_AGENTS_SUCCESS', () => {
  describe('getAgents', () => {
    let agents = [...state.agents];

    test('Receives agents', async () => {
      fetchImplementation(fetch, 200, { agents });
      const returnValue = await store.dispatch(actions.getAgents());
      expect(fetch.mock.calls.length).toBe(1);
      expect(fetch.mock.calls[0]).toMatchInlineSnapshot(`
        Array [
          "/api/agents",
        ]
      `);
      expect(returnValue).toEqual(agents);

      await mockStore.dispatch(actions.getAgents());
      expect(mockStore.getActions()).toMatchSnapshot();
    });

    test('Receives undefined', async () => {
      fetchImplementation(fetch, 200, { agents: undefined });
      const returnValue = await store.dispatch(actions.getAgents());
      expect(fetch.mock.calls.length).toBe(1);
      expect(fetch.mock.calls[0]).toMatchInlineSnapshot(`
        Array [
          "/api/agents",
        ]
      `);
      expect(returnValue).toMatchInlineSnapshot(`Array []`);

      await mockStore.dispatch(actions.getAgents());
      expect(mockStore.getActions()).toMatchSnapshot();
    });
  });
  describe('getAgents, filtered', () => {
    let agents = [...state.agents];

    test('Receives agents', async () => {
      fetchImplementation(fetch, 200, { agents });
      const returnValue = await store.dispatch(actions.getAgents('is_active'));
      expect(fetch.mock.calls.length).toBe(1);
      expect(fetch.mock.calls[0]).toMatchInlineSnapshot(`
        Array [
          "/api/agents",
        ]
      `);
      expect(returnValue).toEqual(agents);

      await mockStore.dispatch(actions.getAgents());
      expect(mockStore.getActions()).toMatchSnapshot();
    });

    test('Receives undefined', async () => {
      fetchImplementation(fetch, 200, { agents: undefined });
      const returnValue = await store.dispatch(actions.getAgents('is_active'));
      expect(fetch.mock.calls.length).toBe(1);
      expect(fetch.mock.calls[0]).toMatchInlineSnapshot(`
        Array [
          "/api/agents",
        ]
      `);
      expect(returnValue).toMatchInlineSnapshot(`Array []`);

      await mockStore.dispatch(actions.getAgents());
      expect(mockStore.getActions()).toMatchSnapshot();
    });
  });
});

describe('GET_AGENTS_ERROR', () => {
  describe('getAgents', () => {
    test('Receives error', async () => {
      fetchImplementation(fetch, 200, { error });
      const returnValue = await store.dispatch(actions.getAgents());
      expect(fetch.mock.calls.length).toBe(1);
      expect(returnValue).toEqual(null);
      await mockStore.dispatch(actions.getAgents());
      expect(mockStore.getActions()).toMatchSnapshot();
    });
  });
});
