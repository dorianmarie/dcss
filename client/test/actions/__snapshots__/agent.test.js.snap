// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET_AGENT_ERROR createAgent Receives an error 2`] = `
Array [
  Object {
    "error": Object {
      "error": [Error: something unexpected happened on the server],
    },
    "type": "GET_AGENT_ERROR",
  },
]
`;

exports[`GET_AGENT_ERROR getAgent Receives an error 2`] = `
Array [
  Object {
    "error": Object {
      "error": [Error: something unexpected happened on the server],
    },
    "type": "GET_AGENT_ERROR",
  },
]
`;

exports[`GET_AGENT_SUCCESS createAgent Empty params 1`] = `Array []`;

exports[`GET_AGENT_SUCCESS createAgent Receives a agent 2`] = `
Array [
  Object {
    "agent": Object {
      "configuration": Object {
        "bar": "2",
        "baz": "c",
        "foo": "false",
      },
      "created_at": "2021-02-25T17:31:33.826Z",
      "deleted_at": null,
      "description": "Detects the presense of an emoji character in your text",
      "endpoint": "ws://emoji-analysis-production.herokuapp.com",
      "id": 1,
      "interaction": Object {
        "created_at": "2021-02-25T15:09:05.001302-05:00",
        "deleted_at": null,
        "id": 18,
        "name": "ChatPrompt",
        "updated_at": null,
      },
      "is_active": true,
      "name": "Fake Agent",
      "owner": Object {
        "email": "super@email.com",
        "id": 999,
        "is_anonymous": false,
        "is_super": true,
        "personalname": "Super User",
        "roles": Array [
          "participant",
          "super_admin",
          "facilitator",
          "researcher",
        ],
        "username": "superuser",
      },
      "self": Object {
        "email": null,
        "id": 148,
        "is_anonymous": true,
        "is_super": false,
        "lastseen_at": "2021-02-25T13:08:57.323-05:00",
        "personalname": "Emoji Analysis",
        "roles": null,
        "single_use_password": false,
        "username": "ebe565050b31cbb4e7eacc39b23e2167",
      },
      "updated_at": "2021-02-25T20:09:04.999Z",
    },
    "type": "GET_AGENT_SUCCESS",
  },
]
`;

exports[`GET_AGENT_SUCCESS getAgent Receives a agent 2`] = `
Array [
  Object {
    "agent": Object {
      "configuration": Object {
        "bar": "2",
        "baz": "c",
        "foo": "false",
      },
      "created_at": "2021-02-25T17:31:33.826Z",
      "deleted_at": null,
      "description": "Detects the presense of an emoji character in your text",
      "endpoint": "ws://emoji-analysis-production.herokuapp.com",
      "id": 1,
      "interaction": Object {
        "created_at": "2021-02-25T15:09:05.001302-05:00",
        "deleted_at": null,
        "id": 18,
        "name": "ChatPrompt",
        "updated_at": null,
      },
      "is_active": true,
      "name": "Fake Agent",
      "owner": Object {
        "email": "super@email.com",
        "id": 999,
        "is_anonymous": false,
        "is_super": true,
        "personalname": "Super User",
        "roles": Array [
          "participant",
          "super_admin",
          "facilitator",
          "researcher",
        ],
        "username": "superuser",
      },
      "self": Object {
        "email": null,
        "id": 148,
        "is_anonymous": true,
        "is_super": false,
        "lastseen_at": "2021-02-25T13:08:57.323-05:00",
        "personalname": "Emoji Analysis",
        "roles": null,
        "single_use_password": false,
        "username": "ebe565050b31cbb4e7eacc39b23e2167",
      },
      "updated_at": "2021-02-25T20:09:04.999Z",
    },
    "type": "GET_AGENT_SUCCESS",
  },
]
`;

exports[`GET_AGENTS_ERROR getAgents Receives error 1`] = `
Array [
  Object {
    "error": Object {
      "error": [Error: something unexpected happened on the server],
    },
    "type": "GET_AGENTS_ERROR",
  },
]
`;

exports[`GET_AGENTS_SUCCESS getAgents Receives agents 2`] = `
Array [
  Object {
    "agents": Array [
      Object {
        "configuration": Object {
          "bar": "2",
          "baz": "c",
          "foo": "false",
        },
        "created_at": "2021-02-25T17:31:33.826Z",
        "deleted_at": null,
        "description": "Detects the presense of an emoji character in your text",
        "endpoint": "ws://emoji-analysis-production.herokuapp.com",
        "id": 1,
        "interaction": Object {
          "created_at": "2021-02-25T15:09:05.001302-05:00",
          "deleted_at": null,
          "id": 18,
          "name": "ChatPrompt",
          "updated_at": null,
        },
        "is_active": true,
        "name": "Emoji Analysis",
        "owner": Object {
          "email": "super@email.com",
          "id": 999,
          "is_anonymous": false,
          "is_super": true,
          "personalname": "Super User",
          "roles": Array [
            "participant",
            "super_admin",
            "facilitator",
            "researcher",
          ],
          "username": "superuser",
        },
        "self": Object {
          "email": null,
          "id": 148,
          "is_anonymous": true,
          "is_super": false,
          "lastseen_at": "2021-02-25T13:08:57.323-05:00",
          "personalname": "Emoji Analysis",
          "roles": null,
          "single_use_password": false,
          "username": "ebe565050b31cbb4e7eacc39b23e2167",
        },
        "updated_at": "2021-02-25T20:09:04.999Z",
      },
    ],
    "type": "GET_AGENTS_SUCCESS",
  },
]
`;

exports[`GET_AGENTS_SUCCESS getAgents Receives undefined 3`] = `
Array [
  Object {
    "agents": Array [],
    "type": "GET_AGENTS_SUCCESS",
  },
]
`;

exports[`GET_AGENTS_SUCCESS getAgents, filtered Receives agents 2`] = `
Array [
  Object {
    "agents": Array [
      Object {
        "configuration": Object {
          "bar": "2",
          "baz": "c",
          "foo": "false",
        },
        "created_at": "2021-02-25T17:31:33.826Z",
        "deleted_at": null,
        "description": "Detects the presense of an emoji character in your text",
        "endpoint": "ws://emoji-analysis-production.herokuapp.com",
        "id": 1,
        "interaction": Object {
          "created_at": "2021-02-25T15:09:05.001302-05:00",
          "deleted_at": null,
          "id": 18,
          "name": "ChatPrompt",
          "updated_at": null,
        },
        "is_active": true,
        "name": "Emoji Analysis",
        "owner": Object {
          "email": "super@email.com",
          "id": 999,
          "is_anonymous": false,
          "is_super": true,
          "personalname": "Super User",
          "roles": Array [
            "participant",
            "super_admin",
            "facilitator",
            "researcher",
          ],
          "username": "superuser",
        },
        "self": Object {
          "email": null,
          "id": 148,
          "is_anonymous": true,
          "is_super": false,
          "lastseen_at": "2021-02-25T13:08:57.323-05:00",
          "personalname": "Emoji Analysis",
          "roles": null,
          "single_use_password": false,
          "username": "ebe565050b31cbb4e7eacc39b23e2167",
        },
        "updated_at": "2021-02-25T20:09:04.999Z",
      },
    ],
    "type": "GET_AGENTS_SUCCESS",
  },
]
`;

exports[`GET_AGENTS_SUCCESS getAgents, filtered Receives undefined 3`] = `
Array [
  Object {
    "agents": Array [],
    "type": "GET_AGENTS_SUCCESS",
  },
]
`;

exports[`SET_AGENT_ERROR setAgent Receives an error 2`] = `
Array [
  Object {
    "error": Object {
      "error": [Error: something unexpected happened on the server],
    },
    "type": "SET_AGENT_ERROR",
  },
]
`;

exports[`SET_AGENT_SUCCESS setAgent Empty params 1`] = `Array []`;

exports[`SET_AGENT_SUCCESS setAgent Receives a agent 2`] = `
Array [
  Object {
    "agent": Object {
      "configuration": Object {
        "bar": "2",
        "baz": "c",
        "foo": "false",
      },
      "created_at": "2021-02-25T17:31:33.826Z",
      "deleted_at": null,
      "description": "Detects the presense of an emoji character in your text",
      "endpoint": "ws://emoji-analysis-production.herokuapp.com",
      "id": 1,
      "interaction": Object {
        "created_at": "2021-02-25T15:09:05.001302-05:00",
        "deleted_at": null,
        "id": 18,
        "name": "ChatPrompt",
        "updated_at": null,
      },
      "is_active": true,
      "name": "Fake Agent",
      "owner": Object {
        "email": "super@email.com",
        "id": 999,
        "is_anonymous": false,
        "is_super": true,
        "personalname": "Super User",
        "roles": Array [
          "participant",
          "super_admin",
          "facilitator",
          "researcher",
        ],
        "username": "superuser",
      },
      "self": Object {
        "email": null,
        "id": 148,
        "is_anonymous": true,
        "is_super": false,
        "lastseen_at": "2021-02-25T13:08:57.323-05:00",
        "personalname": "Emoji Analysis",
        "roles": null,
        "single_use_password": false,
        "username": "ebe565050b31cbb4e7eacc39b23e2167",
      },
      "updated_at": "2021-02-25T20:09:04.999Z",
    },
    "type": "SET_AGENT_SUCCESS",
  },
]
`;

exports[`SET_AGENT_SUCCESS setAgent Receives a agent, updates scenario 2`] = `
Array [
  Object {
    "agent": Object {
      "configuration": Object {
        "bar": "2",
        "baz": "c",
        "foo": "false",
      },
      "created_at": "2021-02-25T17:31:33.826Z",
      "deleted_at": null,
      "description": "Detects the presense of an emoji character in your text",
      "endpoint": "ws://emoji-analysis-production.herokuapp.com",
      "id": 1,
      "interaction": Object {
        "created_at": "2021-02-25T15:09:05.001302-05:00",
        "deleted_at": null,
        "id": 18,
        "name": "ChatPrompt",
        "updated_at": null,
      },
      "is_active": true,
      "name": "Fake Agent",
      "owner": Object {
        "email": "super@email.com",
        "id": 999,
        "is_anonymous": false,
        "is_super": true,
        "personalname": "Super User",
        "roles": Array [
          "participant",
          "super_admin",
          "facilitator",
          "researcher",
        ],
        "username": "superuser",
      },
      "self": Object {
        "email": null,
        "id": 148,
        "is_anonymous": true,
        "is_super": false,
        "lastseen_at": "2021-02-25T13:08:57.323-05:00",
        "personalname": "Emoji Analysis",
        "roles": null,
        "single_use_password": false,
        "username": "ebe565050b31cbb4e7eacc39b23e2167",
      },
      "updated_at": "2021-02-25T20:09:04.999Z",
    },
    "type": "SET_AGENT_SUCCESS",
  },
]
`;
